components/apimgt/org.wso2.carbon.apimgt.spec.parser/src/main/java/org/wso2/carbon/apimgt/spec/parser/definitions/asyncapi/AsyncApiParseOptions.java

```diff
@@ -0,0 +1,40 @@
+/*
+ *   Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com)
+ *
+ *   WSO2 LLC. licenses this file to you under the Apache License,
+ *   Version 2.0 (the "License"); you may not use this file except
+ *   in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+package org.wso2.carbon.apimgt.spec.parser.definitions.asyncapi;
+
+/**
+ * Holds configuration options used during AsyncAPI parsing.
+ * This class is primarily used to determine which version of the AsyncAPI v2 parser
+ * should be used (default/new or legacy).
+ */
+public class AsyncApiParseOptions {
+
+    private boolean defaultAsyncApiParserVersion = true;
```

Comment:
the variable name and getter and setter for the config is not intuitive. Shall we change it to reflect the configuration name.